launch.go

```diff
@@ -72,6 +73,32 @@ func init() {
 	client = retryablehttp.NewClient()
 }
 
+/* has HTTP or HTTPS protocol
+targetURL => URL
+*/
+func hasHTTPProtocol(targetURL *url.URL) bool {
+	return targetURL.Scheme == "http" || targetURL.Scheme == "https"
+}
+
+/* make pushgateway url
+baseURL => base url for pushgateway
+buildID => sd build id
+*/
+func makePushgatewayURL(baseURL string, buildID int) (*url.URL, error) {
+	var pushgatewayURL string = baseURL
+	u, err := url.Parse(pushgatewayURL)
+	if err != nil {
+		log.Printf("makePushgatewayURL: failed to parse url [%v], buildId:[%v], error:[%v]", pushgatewayURL, buildID, err)
+		return nil, err
+	}
+
+	if !hasHTTPProtocol(u) {
+		return nil, errors.New("Pushgateway url has no http/https protocol. Please make sure it.")
+	}
+	u.Path = u.Path + "/metrics/job/containerd/instance/" + strconv.Itoa(buildID)
```

Comment:
Is it better to use path.Join?

If baseURL is like 'http://sample.com/', u.Path will be 'http://sample.com//metrics/job/containerd/instance/id'

Reply:
I fixed it.